File: src/main/java/no/nav/arbeidsgiver/min_side/services/digisyfo/NarmesteLederHendelse.java
package no.nav.arbeidsgiver.min_side.services.digisyfo;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.UUID;

@JsonIgnoreProperties(ignoreUnknown = true)
public class NarmesteLederHendelse {
    public final UUID narmesteLederId;
    public final String narmesteLederFnr;
    public final String aktivTom;

    @JsonCreator(mode = JsonCreator.Mode.PROPERTIES)
    public NarmesteLederHendelse(
            @JsonProperty("narmesteLederId") UUID narmesteLederId,
            @JsonProperty("narmesteLederFnr") String narmesteLederFnr,
            @JsonProperty("aktivTom") String aktivTom
    ) {
        this.narmesteLederId = narmesteLederId;
        this.narmesteLederFnr = narmesteLederFnr;
        this.aktivTom = aktivTom;
    }
}

File: src/main/kotlin/no/nav/arbeidsgiver/min_side/services/digisyfo/NarmesteLederHendelse.kt
package no.nav.arbeidsgiver.min_side.services.digisyfo

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import no.nav.arbeidsgiver.min_side.infrastruktur.SerializableUUID

@Serializable
class NarmesteLederHendelse(
    @SerialName("narmesteLederId")  val narmesteLederId: SerializableUUID,
    @SerialName("narmesteLederFnr") val narmesteLederFnr: String,
    @SerialName("aktivTom") val aktivTom: String? = null,
    @SerialName("orgnummer") val virksomhetsnummer: String,
    @SerialName("fnr") val ansattFnr: String
)